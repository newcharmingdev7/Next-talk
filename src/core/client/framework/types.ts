// TODO: (@cvle) Extract useful common types into its own package.
export { Omit, Overwrite, PropTypesOf } from "coral-ui/types";
export { DeepPartial } from "coral-common/types";

export type RelayEnumLiteral<T> = keyof T | "%future added value";
