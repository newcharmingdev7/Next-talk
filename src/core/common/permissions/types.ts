export type UserRole =
  | "%future added value"
  | "COMMENTER"
  | "MEMBER"
  | "STAFF"
  | "MODERATOR"
  | "ADMIN";

export interface User {
  id: string;
  role: UserRole;
  moderationScopes?: null | {
    scoped?: boolean;
    siteIDs?: string[];
  };
}

export type Moderator = User & {
  role: "MODERATOR";
};

export type LTEModerator = User & {
  role: "MODERATOR" | "STAFF" | "MEMBER" | "COMMENTER";
};

export const isModerator = (user: User): user is Moderator => {
  return user.role === "MODERATOR";
};

export const isLTEModerator = (user: User): user is LTEModerator => {
  return user.role !== "ADMIN";
};
