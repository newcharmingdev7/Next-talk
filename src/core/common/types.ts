export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
  { [P in U]: never } & { [x: string]: never })[T];

export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;

export type Overwrite<T, U> = Pick<T, Diff<keyof T, keyof U>> & U;

export type Sub<T, U> = Pick<T, Diff<keyof T, keyof U>>;

/**
 * Make all properties in T writeable
 */
export type Writeable<T> = { -readonly [P in keyof T]: T[P] };

export type Promiseable<T> = Promise<T> | T;
