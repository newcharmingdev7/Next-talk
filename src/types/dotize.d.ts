declare module "dotize" {
  export function convert(obj: Object): Record<string, any>
}
