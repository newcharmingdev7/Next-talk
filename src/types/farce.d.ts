declare module "farce" {
  export class BrowserProtocol {}
  export class queryMiddleware {}
}
