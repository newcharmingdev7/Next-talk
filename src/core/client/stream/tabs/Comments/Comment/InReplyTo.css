$commentsInReplyToUsernameColor: var(--palette-text-500);
$commentsInReplyToLabelColor: var(--palette-text-100);
$commentsInReplyToIconColor: var(--palette-text-100);

button {
  background-color: inherit;
  border: none;
  padding: 0;
  margin: 0;
}

.icon {
  color: $commentsInReplyToIconColor;
}

.username {
  font-family: var(--font-family-secondary);
  font-weight: var(--font-weight-secondary-bold);
  font-size: var(--font-size-3);
  line-height: 1;

  color: $commentsInReplyToUsernameColor;
}

.inReplyTo {
  font-family: var(--font-family-primary);
  font-weight: var(--font-weight-primary-semi-bold);
  font-size: var(--font-size-2);
  line-height: 1;
  color: $commentsInReplyToLabelColor;
}
