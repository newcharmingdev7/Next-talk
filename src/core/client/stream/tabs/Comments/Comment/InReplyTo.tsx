import { Localized } from "@fluent/react/compat";
import cn from "classnames";
import React, { FunctionComponent } from "react";

import { useCoralContext } from "coral-framework/lib/bootstrap";
import CLASSES from "coral-stream/classes";
import computeCommentElementID from "coral-stream/tabs/Comments/Comment/computeCommentElementID";
import { Flex, Icon } from "coral-ui/components/v2";

import { CommentContainer_comment as CommentData } from "coral-stream/__generated__/CommentContainer_comment.graphql";

import styles from "./InReplyTo.css";

interface Props {
  parent: CommentData["parent"];
  isLink: boolean;
}

const InReplyTo: FunctionComponent<Props> = ({ parent }) => {
  const { pym } = useCoralContext();


  const navigateToParent = (id: string) => {
    const elemId = computeCommentElementID(id);
    const elem = document.getElementById(elemId);
    if (elem) {
      void pym?.scrollParentToChildEl(elemId);
      elem.focus();
      (elem as any).style["background-color"] = "pink";
    } else {
      /* eslint-disable-next-line */
      console.log("TODO: its not on the page~");
    }
  };

  const Username = () => (
    <button
      onClick={() => navigateToParent(parent!.id)}
      className={cn(styles.username, CLASSES.comment.inReplyTo.username)}
    >
      {parent!.author!.username}
    </button>
  );

  return (
    <Flex alignItems="center" className={CLASSES.comment.inReplyTo.$root}>
      <Icon className={styles.icon}>reply</Icon>{" "}
      <Localized id="comments-inReplyTo" Username={<Username />}>
        <span className={cn(styles.inReplyTo, CLASSES.comment.inReplyTo.text)}>
          {"In reply to <Username></Username>"}
        </span>
      </Localized>
    </Flex>
  );
};

export default InReplyTo;
