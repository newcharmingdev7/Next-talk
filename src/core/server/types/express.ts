import { Request } from "express";

import { Tenant } from "talk-server/models/tenant";
import { User } from "talk-server/models/user";
import TenantCache from "talk-server/services/tenant/cache";

export interface Request extends Request {
  user?: User;
  tenant?: Tenant;
  tenantCache: TenantCache;
}
