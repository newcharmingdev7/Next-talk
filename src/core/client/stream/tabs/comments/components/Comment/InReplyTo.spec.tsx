import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import InReplyTo from "./InReplyTo";

it("renders correctly", () => {
  const props: PropTypesOf<typeof InReplyTo> = {
    username: "Username",
  };
  const wrapper = shallow(<InReplyTo {...props} />);
  expect(wrapper).toMatchSnapshot();
});
