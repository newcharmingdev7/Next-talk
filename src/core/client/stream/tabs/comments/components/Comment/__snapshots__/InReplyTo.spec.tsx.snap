// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<withPropsOnChange(Flex)
  alignItems="center"
>
  <withPropsOnChange(Icon)
    className="InReplyTo-icon"
  >
    reply
  </withPropsOnChange(Icon)>
   
  <Localized
    id="comments-inReplyTo"
    username={<Username />}
  >
    <withPropsOnChange(Typography)
      className="InReplyTo-inReplyTo"
      container="span"
      variant="timestamp"
    >
      In reply to &lt;username&gt;&lt;username&gt;
    </withPropsOnChange(Typography)>
  </Localized>
</withPropsOnChange(Flex)>
`;
